jobs/integration/tigera_aws.py

```diff
@@ -149,19 +145,28 @@ def bootstrap():
                 vpc["Ipv6CidrBlockAssociationSet"],
             )
         )
+
+    # AWS gives us a larger CIDR block than we can use directly, so we have to chop it down
+    # "A subnet's IPv6 CIDR block is a fixed prefix length of /64."
+    # https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html#vpc-sizing-ipv6
+    ipv6_network = ipaddress.IPv6Network(ipv6_cidr_block)
+    ipv6_subnets = ipv6_network.subnets(64 - ipv6_network.prefixlen)
+    ipv6_subnet_cidrs = [str(subnet) for subnet in ipv6_subnets]
+
     # Must be done in separate requests per doc
     ec2.modify_vpc_attribute(VpcId=vpc_id, EnableDnsHostnames={"Value": True})
     ec2.modify_vpc_attribute(VpcId=vpc_id, EnableDnsSupport={"Value": True})
 
     # Create subnets
     num_subnets = os.environ.get("NUM_SUBNETS")
     num_subnets = int(num_subnets) if num_subnets else 1
-    subnet_cidrs = SUBNET_CIDRS[:num_subnets]
-    for subnet_cidr in subnet_cidrs:
+    for i in range(num_subnets):
+        subnet_cidr = SUBNET_CIDRS[i]
+        subnet_ipv6_cidr = ipv6_subnet_cidrs[i]
         subnet_id = ec2.create_subnet(
             VpcId=vpc_id,
             CidrBlock=subnet_cidr,
-            Ipv6CidrBlock=ipv6_cidr_block,
+            Ipv6CidrBlock=subnet_ipv6_cidr,
```

Comment:
This fixes an issue where the code tried to create multiple subnets with the same Ipv6CidrBlock, resulting in failure due to conflict. Now each subnet will get a unique IPv6 CIDR block.